src/extensionConflicts.ts

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved..
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { commands, Extension, extensions, window } from 'vscode';
+
+// A set of VSCode extension ID's that conflict with our extension
+const conflictingIDs = [
+  'haaaad.ansible',
+  'sysninja.vscode-ansible-mod',
+  'tomaciazek.ansible',
+  'vscoss.vscode-ansible',
+  'zbr.vscode-ansible',
+];
+
+// A set of VSCode extension ID's that are currently uninstalling
+const uninstallingIDs = new Set();
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isExtensionPresent(obj: any): obj is Extension<any> {
+  return obj !== undefined && !uninstallingIDs.has(obj.id);
+}
+
+/**
+ * Get all of the installed extensions that currently conflict with VSCode-YAML
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function getConflictingExtensions(): Extension<any>[] {
+
+  return conflictingIDs.map((x) => extensions.getExtension(x))
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    .filter<Extension<any>>((ext): ext is Extension<any> => isExtensionPresent(ext));
+}
+
+/**
+ * Display the uninstall conflicting extension notification if there are any conflicting extensions currently installed
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function showUninstallConflictsNotification(conflictingExts: Extension<any>[]): void {
+  // Add all available conflicting extensions to the uninstalling IDs map
+  for (const ext of conflictingExts) {
+    uninstallingIDs.add(ext.id);
+  }
+
+  const uninstallMsg = 'Uninstall';
+
+  // Gather all the conflicting display names
+  let conflictMsg = '';
+  if (conflictingExts.length === 1) {
+    conflictMsg = `${conflictingExts[0].packageJSON.displayName} extension is incompatible with redhat.ansible. Please uninstall it.`;
+  } else {
+    const extNames: string[] = conflictingExts.map(ext => ext.packageJSON.displayName);
+    conflictMsg = `The ${extNames.join(', ')} extensions are incompatible with redhat.ansible. Please uninstall them.`;
+  }
+
+  if (!conflictingExts.length) {
```

Comment:
This guard expression should probably be the first thing in the function.

Reply:
done